app/clients/letter/dvla.py

```diff
@@ -80,5 +82,59 @@ def authenticate(self):
 
         return response.json()["id-token"]
 
+    def reset_api_key(self):
+        response = self.request.post(
+            f"{current_app.config['DVLA_API_BASE_URL']}/thirdparty-access/v1/new-api-key",
+            headers={
+                "x-api-key": self.dvla_api_key.get(),
+                "Authorization": self.jwt_token,
+            },
+            json={
+                "userName": self.dvla_username.get(),
+                "password": self.dvla_password.get(),
+            },
+        )
+        response.raise_for_status()
+
+        self.dvla_api_key.set(response.json()["newApiKey"])
+
+    def reset_password(self):
+        new_password = self._generate_password()
+        response = self.request.post(
+            f"{current_app.config['DVLA_API_BASE_URL']}/thirdparty-access/v1/password",
+            json={
+                "userName": self.dvla_username.get(),
+                "password": self.dvla_password.get(),
+                "newPassword": new_password,
+            },
+        )
+
+        response.raise_for_status()
+
+        self.dvla_password.set(new_password)
+
+    @staticmethod
+    def _generate_password():
+        """
+        DVLA api password must be at least 8 characters in length and contain upper, lower, numerical and special
+        characters.
+
+        This function creates a valid password of length 34 characters. (30)
```

Comment:
👀

Reply:
whoops fat-thumbed it. will squash out to say just 30 and not spread fake news about.